### PR TITLE
Storage: Correctly highlight dangerous menu items in red 

### LINKED ISSUE #18415: storage: Filesystem delete entry in dropdown's should shown as dangerous
### Explain what happens

For example the stratis filesystem delete dropdown or a normal disk's delete dropdown entry is not shown as dangerous as required per [Patternfly guidelines](https://www.patternfly.org/v4/guidelines/colors/#status-and-state-colors).

image

They should be shown as the image below, with red text.

image

## Reproduce

```
./test/image-prepare fedora-37
./bots/vm-run -s cockpit.socket fedora-37
```

* Open a browser http://localhost:9090 login and open the storage page.
* Click on the `/dev/vda2` filesystem 
* Click on the kebab menu (three dots on the right)
* This should open a dropdown menu with a Delete entry which doesn't have a red text.

### Version of Cockpit

main

### Where is the problem in Cockpit?

None

### Server operating system

None

### Server operating system version

_No response_

### What browsers are you using?

_No response_

### System log

_No response_

### PR DESCRIPTION
**Pull request by MahmoudHamdy02**
Closes #18415.

Dangerous menu items, such as 'Delete', were not highlighted in a red color correctly.

I added a 'danger' property to the `tab_menu_danger_actions` array, to identify the items in `StorageMenuItem`.

Here is a screenshot:
image

I think this makes the separate 'danger' array redundant, since they get merged into one array later. It could simply be one array with a 'danger' property set to false/true. I'm not sure if that would break the separator line placement, but I think placing it afterwards when the array is full won't be an issue.

Please let me know what changes I should make to this PR, I'm open to trying a different approach if needed. Thanks!

### CODE REVIEW

**Review by garrett — CHANGES_REQUESTED** [review]
Why isn't this using PatternFly's `isDanger` menu attribute?

https://www.patternfly.org/v4/components/menu/#danger-menu-item

(We shouldn't be using custom code for things PatternFly provides for us. The CSS instantly stood out to me; we used to have to do things like this a long time ago, but PF added an `isDanger` attribute for us, at our request, a while back.)

**Comment by KKoukiou** [comment]
> Why isn't this using PatternFly's `isDanger` menu attribute?
> 
> https://www.patternfly.org/v4/components/menu/#danger-menu-item
> 
> (We shouldn't be using custom code for things PatternFly provides for us. The CSS instantly stood out to me; we used to have to do things like this a long time ago, but PF added an `isDanger` attribute for us, at our request, a while back.)

Because this uses DropdownItem and not the Menu. This is the only available approach with this component.

**Comment by KKoukiou** [comment]
@MahmoudHamdy02 can you please remove the merge commit?

File: pkg/storaged/storage-controls.jsx [hunk 1]
```diff
@@ -215,8 +215,8 @@ export const StorageUsageBar = ({ stats, critical, block, offset, total, small }
         </div>);
 };
 
-export const StorageMenuItem = ({ onClick, onlyNarrow, children }) => (
-    <DropdownItem className={onlyNarrow ? "show-only-when-narrow" : null}
+export const StorageMenuItem = ({ onClick, onlyNarrow, danger, children }) => (
+    <DropdownItem className={(onlyNarrow ? "show-only-when-narrow" : null) + " " + (danger ? "delete-resource-red" : null)}
```
Comment (KKoukiou) [inline pkg/storaged/storage-controls.jsx:219]: Because javascript is weird this is not correct entiretly :)

Comment (KKoukiou) [reply pkg/storaged/storage-controls.jsx:219]: `"" + null` will result to "null"

For this reason you need to replace null with '' and  you can also remove the middle + " " + spacer, but just adding a space in the " delete-resource-red"

Comment (MahmoudHamdy02) [reply pkg/storaged/storage-controls.jsx]: Hello @KKoukiou!

I've removed the merge commit and changed the null condition, let me know if I should make any more changes, thanks :)

**Review by KKoukiou — COMMENTED** [review]
Just realized we have missed one more occurance of the 'Delete' button;
These also need the new danger property;

pkg/storaged/stratis-details.jsx:        menuitems.push(<StorageMenuItem key="del" onClick={delete_fsys}>{_("Delete")}</StorageMenuItem>);

**Comment by KKoukiou** [comment]
@garrett should the 'Format' action also be red? @MahmoudHamdy02 < let's take the designers opinion.

**Comment by MahmoudHamdy02** [comment]
> pkg/storaged/stratis-details.jsx: menuitems.push({_("Delete")});

After some googling I found that stratis is a storage management library, but it's only for Arch and Fedora. I'm using Ubuntu 22.04. Is there a way to reach the stratis page without having the package installed? Thanks :P

**Comment by MahmoudHamdy02** [comment]
Hi @KKoukiou 

I had some trouble running the test VM through cockpit, so I installed a full fedora 37 through Ubuntu's KVM.

I fixed the delete button color in the `stratis-details` page, let me know if I need to make more changes!

**Comment by garrett** [comment]
> should the 'Format' action also be red? @MahmoudHamdy02 < let's take the designers opinion.

Thanks for asking! Formatting is "dangerous" as data loss will definitely happen, so: Yes.

**Review by KKoukiou — APPROVED** [review]
Commit nees squashing on merge. Otherwise all great. thanks!

**Comment by martinpitt** [comment]
Thank you @MahmoudHamdy02 !

**Pull request merged**